src/main/java/edu/wpi/grip/ui/OperationView.java

```diff
@@ -88,5 +88,31 @@ public void initialize(URL url, ResourceBundle resourceBundle) {
             Step step = new Step(this.eventBus, this.operation);
             this.eventBus.post(new StepAddedEvent(step));
         });
+
+        // Ensures that when this element is hidden that it also removes its size calculations
+        this.managedProperty().bind(this.visibleProperty());
+    }
+
+    // Cache the lower case name so that it isn't calculated each time the search bar is used
+    private String nameLoweCase = null;
+    /**
+     * Uses the search string from the palette and allows this element to determine if it should be visible
+     * @param lowerCaseSearchString The search string that should be used to determine visibility
+     */
+    void determineVisibility(String lowerCaseSearchString){
+        assert lowerCaseSearchString.equals(lowerCaseSearchString.toLowerCase()) : "The string passed to determine visibility was not lower case";
```

Comment:
A more robust solution might be to accept any string and do a `toLowerCase()` inside this method.  I very much doubt that would have the slightest performance impact.